server/src/routes/routes.ts

```
@@ -548,3 +552,34 @@ export async function fetchMapImages(
     reply.status(500).send({ error: "Internal Server Error" });
   }
 }
+
+export const fetchWorkshopCoverImage = async function (
+  request: FastifyRequest<{ Params: { workshopId: number } }>,
+  reply: FastifyReply
+) {
+  const { workshopId } = request.params;
+  const defaultFilepath = path.join(assetconfigs.datapackImagesDirectory, "TSCreatorLogo.png");
+  try {
+    const imageFilepath = await fetchWorkshopCoverPictureFilepath(workshopId);
+    if (!imageFilepath) {
+      if (!(await checkFileExists(defaultFilepath))) {
+        reply.status(404).send({ error: "Default image not found" });
+        return;
+      }
+      reply.send(await readFile(defaultFilepath));
+      return;
```

this is the same code as in the `catch` block, so i think it should just throw an error here